feat: add `iota-rest-kv` docker configuration (#5299)
iotaledger/iota: dev-tools/iota-rest-kv/README.md

# Kv Store Rest Api

> This Docker container enables reading historical data from the kv store (e.g., `KvStoreWorker` from the `iota-data-ingestion` crate writes historical data into remote storages and database). The service requires AWS credentials for proper execution.

Reviewer: Nit: Phrasing
```suggestion
This docker-compose configuration allows running the `iota-rest-kv` service.  The service requires AWS credentials for proper execution.
```

> ## Configuration
> 
> The application to function does require a config file.
> 
> ### Configuration File
> 
> A default configuration file is provided at `config/config.yaml`. This configuration can be customized based on your needs and is mounted into the container via Docker Compose.

Reviewer: Nit: Consolidate content
```suggestion
## Configuration

The application requires a `yaml` file for its configuration.

A default configuration file is provided at `config/config.yaml`. This configuration can be customized based on your needs and is mounted into the container via Docker Compose.
```

```yaml
# Remote KV Store REST API config
#
# The Rest Api address
rest-api-address: "0.0.0.0:3555"

# remote Object Store config for more info:
# - https://docs.iota.org/operator/archives#set-up-archival-fallback
#
# The object store config should point to the same object store the
# `KvStoreWorker` from the `iota-data-ingestion` crate uses to write data
object-store-config:
  object-store: "S3"
  aws-endpoint: "http://0.0.0.0:4566"
  bucket: "iota-storage-bucket"
  aws-access-key-id: "test"
  aws-secret-access-key: "test"
  aws-allow-http: true
  object-store-connection-limit: 20

# DynamoDb config
#
# The DynamoDb config should point to the same tables the
# `KvStoreWorker` from the `iota-data-ingestion` crate uses to write data
dynamo-db-config:
  aws-access-key-id: "test"
  aws-secret-access-key: "test"
  aws-region: "us-east-1"
  aws-endpoint: "http://0.0.0.0:4566"
  # DynamoDB table name
  table-name: "iota-storage"
```

## Usage

#### 1. Build the required image

```shell

> pushd <iota project directory>/docker/iota-rest-kv && ./build.sh && popd

Reviewer 2: Have you considered making the directory switch in the script itself?
We have few scripts that detect repo root like:
```
REPO_ROOT=$(git rev-parse --show-toplevel)
```
and then execute stuff relative to the detected path. This could make the README instructions simpler, and the script easier to use overall.

Author: @tomxey I'm afraid this change will get out the scope of this issue. 

The` build.sh` script internally calls an helper script, `${REPO_ROOT}/docker/utils/build-script.sh`, which assumes the `Dockerfile` is located in the current directory. To accommodate this assumption, we must manually change the current directory to the location of the `Dockerfile` before invoking the `build.sh` script. 

Implementing this change will affect other parts of the project, we could do this in a separate issue for sure.
